modules/cluster/cloud-config.yml

```diff
@@ -1,112 +1,111 @@
+            },
+            "append_dimensions": {
+                "ImageId": "$${aws:ImageId}",
+                "InstanceId": "$${aws:InstanceId}",
+                "InstanceType": "$${aws:InstanceType}",
+                "AutoScalingGroupName": "$${aws:AutoScalingGroupName}"
+            },
+            "aggregation_dimensions": [["InstanceId"], ["AutoScalingGroupName"]]
+        },
+        "logs": {
+            "logs_collected": {
+                "files": {
+                    "collect_list": [
+                        {
+                            "file_path": "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log*",
+                            "log_group_name": "${log_group_name}",
+                            "log_stream_name": "{instance_id}/amazon-cloudwatch-agent",
+                            "timezone": "UTC"
+                        },
+                        {
+                            "file_path": "/var/log/ecs/ecs-agent.log*",
+                            "log_group_name": "${log_group_name}",
+                            "log_stream_name": "{instance_id}/amazon-ecs-agent",
+                            "timezone": "UTC"
+                        }
+                    ]
+                }
+            },
+            "log_stream_name": "{instance_id}/untitled-logs"
+        }
+    }
 
-  - path: "/etc/ecs/ecs.config"
-    permissions: "0744"
-    owner: "root"
-    content: |
-      ECS_CLUSTER=${ecs_cluster_name}
-      ECS_LOG_LEVEL=${ecs_log_level}
-      ECS_ENABLE_CONTAINER_METADATA=true
-      ECS_ENABLE_TASK_IAM_ROLE=true
-      ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
-      ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
+- path: "/etc/ecs/ecs.config"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    ECS_CLUSTER=${ecs_cluster_name}
+    ECS_LOG_LEVEL=${ecs_log_level}
+    ECS_ENABLE_CONTAINER_METADATA=true
+    ECS_ENABLE_TASK_IAM_ROLE=true
+    ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
+    ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
 
-  - path: "/usr/local/scripts/cloudformation-signal.sh"
-    permissions: "0744"
-    owner: "root"
-    content: |
-      #! /bin/bash
-
-      set -euo pipefail
-
-      function await_process() {
-        echo -n "Waiting for $1..."
-        while ! pgrep -f "$1" > /dev/null; do
-          sleep 1
-        done
-        echo "Done!"
-      }
-      await_process "/usr/libexec/amazon-ecs-init start"
+- path: "/usr/local/scripts/cloudformation-signal.sh"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    #! /bin/bash
+    set -uo pipefail
+    INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+    echo -n "Waiting for Instance $INSTANCE_ID to be ACTIVE in cluster ${ecs_cluster_name}"
+    function await_process() {
+      CHECK_ACTIVE=$(aws --region ${region} ecs list-container-instances --cluster ${ecs_cluster_name} --status ACTIVE --filter ec2InstanceId==$INSTANCE_ID | grep 'arn:aws:ecs')
+      while [ -z $CHECK_ACTIVE ]; do
```

Comment:
If you just initialise `CHECK_ACTIVE` as an empty string the `while [ -z $CHECK_ACTIVE ]` should still enter the loop; just have to remember to move the sleep. Bonus of doing this is that you don't have to repeat the command, and change it in two places when it needs a change.